pkg/resources/grant_ownership.go

```diff
@@ -0,0 +1,668 @@
+	} else {
+		accountRoleName, err := client.ContextFunctions.CurrentRole(ctx)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		err = client.Grants.GrantOwnership( // TODO: Should we always set outbound privileges to COPY in delete operation or set it to the config value?
+			ctx,
+			grantOn,
+			sdk.OwnershipGrantTo{
+				AccountRoleName: sdk.Pointer(sdk.NewAccountObjectIdentifier(accountRoleName)), // TODO: What if current role is database role (not a common but possible case)
+				// DatabaseRoleName: TODO: handle later
+			},
+			getOwnershipGrantOpts(id),
+		)
+		if err != nil {
+			return diag.Diagnostics{
+				diag.Diagnostic{
+					Severity: diag.Error,
+					Summary:  "An error occurred when transferring ownership back to the original role",
+					Detail:   fmt.Sprintf("Id: %s\nError: %s", d.Id(), err),
+				},
+			}
+		}
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+func ReadGrantOwnership(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	id, err := ParseGrantOwnershipId(d.Id())
+	if err != nil {
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "Failed to parse internal identifier",
+				Detail:   fmt.Sprintf("Id: %s\nError: %s", d.Id(), err),
+			},
+		}
+	}
+
+	opts, grantedOn := prepareShowGrantsRequestForGrantOwnership(id)
+	if opts == nil {
+		return nil
+	}
+
+	client := meta.(*provider.Context).Client
+
+	grants, err := client.Grants.Show(ctx, opts)
+	if err != nil {
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "Failed to retrieve grants",
+				Detail:   fmt.Sprintf("Id: %s\nError: %s", d.Id(), err),
+			},
+		}
+	}
+
+	ownershipFound := false
+
+	for _, grant := range grants {
+		if grant.Privilege != "OWNERSHIP" {
+			continue
+		}
+
+		// Future grants do not have grantedBy, only current grants do.
+		// If grantedby is an empty string, it means terraform could not have created the grant
+		if (opts.Future == nil || !*opts.Future) && grant.GrantedBy.Name() == "" {
+			continue
+		}
+
+		// grant_on is for future grants, granted_on is for current grants.
+		// They function the same way though in a test for matching the object type
+		if grantedOn != grant.GrantedOn && grantedOn != grant.GrantOn {
+			continue
+		}
+
+		switch id.GrantOwnershipTargetRoleKind {
+		case ToAccountGrantOwnershipTargetRoleKind:
+			if grant.GranteeName.Name() == id.AccountRoleName.Name() {
+				ownershipFound = true
+			}
+		case ToDatabaseGrantOwnershipTargetRoleKind:
+			if grant.GranteeName.Name() == id.DatabaseRoleName.Name() {
+				ownershipFound = true
+			}
+		}
+	}
+
+	if !ownershipFound {
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "Couldn't find OWNERSHIP privilege on target object",
+				Detail:   fmt.Sprintf("Id: %s", d.Id()),
+			},
+		}
+	}
+
+	return nil
+}
+
+func getOnObjectIdentifier(objectType sdk.ObjectType, objectName string) (sdk.ObjectIdentifier, error) {
```

Comment:
this looks lika a general use validation method that could be inside the SDK (so each object has its dedicated type of identifier). Please leave a TODO with identifiers rework issue number and add reference to this method inside the issue's description, so that we don't forget about it.